text/0000-int-guidelines.md

```diff
@@ -0,0 +1,164 @@
+- Start Date: 2015-01-26
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+
+This RFC proposes usage guidelines for the various integer types.
+
+# Motivation
+
+The goal of this RFC is to help people decide what integer types to use when they need
+to make a decision for a new API.
+
+It builds on [https://github.com/rust-lang/rfcs/pull/560](the integer overflow RFC),
+which provides debug-time assertions for overflow and underflow. One of the goals
+of this RFC is to provide guidance for when to use unsigned types, and these
+assertions affect the traditional tradeoffs about unsigned types, which are described
+under the detailed design below.
+
+It also draws inspiration from the [Google C++ Style Guide](http://google-styleguide.googlecode.com/svn/trunk/cppguide.html#Integer_Types)
+
+This RFC attempts to balance several concerns:
+
+* Rust developers should have an easy, go-to heuristic for deciding what
+  integer sizes to use when they need to use a number type.
+* 32-bit integers are considerably faster than 64-bit integers in some
+  situations, so using 64-bit integers for tiny numbers, especially in
+  hot code, can result in unnecessarily slower programs.
+* Reflexive use of 32-bit integers too-often results in overflows when the
```

Comment:
So does "reflexive" use of 64-bit integers. It happens to hide overflows in common cases, but it's the edge cases that are hard to find in testing which are the most problematic. You're only exasperating the problem by suggesting that intuition should be used to make this decision.